File: 백준/Silver/5397. 키로거/키로거.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            String key = br.readLine();

            Stack<Character> left = new Stack<>();
            Stack<Character> right = new Stack<>();

            for (char c : key.toCharArray()) {
                if (c == '<') {
                    if (!left.isEmpty()) {
                        right.push(left.pop());
                    }
                } else if (c == '>') {
                    if (!right.isEmpty()) {
                        left.push(right.pop());
                    }
                } else if (c == '-') {
                    if (!left.isEmpty()) {
                        left.pop();
                    }
                } else {
                    left.push(c);
                }
            }

            // 오른쪽에 값이 남아있다면 왼쪽으로 넣어주기
            while(!right.isEmpty()) {
                left.push(right.pop());
            }

            StringBuffer sb = new StringBuffer();
            for (char c : left) {
                sb.append(c);
            }

            bw.write(sb.toString() + "\n");
            bw.flush();
        }
    }
}

File: 백준/Silver/5397. 키로거/키로거.py
t = int(input())

command = ["<", ">", "-"]
for _ in range(t):
    key = input()

    # Cursor를 기점으로 왼쪽 배열과 오른쪽 배열을 분리
    left = []
    right = []
    for i in key:
        if i not in command:
            left.append(i)  # 커서는 값을 입력하고 다음 값으로
        else:
            if i == "<":
                if left:
                    right.append(left.pop())
            elif i == ">":
                if right:
                    left.append(right.pop())
            else:
                if left:
                    left.pop()

    left.extend(reversed(right))
    print("".join(left))
